src/uct/ib/rc/verbs/rc_verbs_ep.c

```diff
@@ -844,6 +850,15 @@ static UCS_CLASS_CLEANUP_FUNC(uct_rc_verbs_ep_t)
                                                  uct_rc_verbs_iface_t);
     uct_worker_progress_remove(iface->super.super.super.worker,
                                &iface->super.super.super.prog);
+    /* NOTE: usually, ci == pi here, but if user calls
+     *       flush(UCT_FLUSH_FLAG_CANCEL) then ep_destroy without next progress,
```

Comment:
can we return credits during flush(UCT_FLUSH_FLAG_CANCEL)?

Reply:
I do not think so, because we emulate flush-cancel in SW but posted to HW operations may be completed before 
1. ep_destroy
2. user posts new operation on any (this or any other) EP 

Then we will have CQ overrun or need to implement extra logic to handle this. The last statement means increase of EP size. In current implementation, handler of canceled TX CQEs will return credits on existing EPs but does not call completion callbacks, they are purged.

Reply:
ok